### PR TITLE
Static pages

### PR DESCRIPTION
**Pull request by weilingooi**
ready

### CODE REVIEW

**Comment by weilingooi** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/* universal */
+
+body {
+    padding-top: 60px;
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/custom.scss]: please tab 2 space

Comment (ngocvt-0484) [inline app/assets/stylesheets/custom.scss]:
```suggestion
  padding-top: 60px;
```

File: app/assets/stylesheets/static_pages.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
+// They will automatically be included in application.css.
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/static_pages.scss]: remove empty file you have created

File: app/helpers/static_pages_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (ngocvt-0484) [inline app/helpers/static_pages_helper.rb]: remove empty file you have created

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb]:
```suggestion
  def full_title page_title
```

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -10,6 +10,10 @@
   </head>
 
   <body>
-    <%= yield %>
-  </body>
-</html>
+  <%= render "layouts/header" %> 
+  <div class="container">
+    <%= yield %> 
+	</div>
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb]:
```suggestion
   </div>
```
need fix format your html
example https://www.granneman.com/webdev/coding/formatting-and-indenting-your-html

File: app/views/static_pages/help.html.erb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<%= provide :title, "Help" %>
+
+
+<h1>Help</h1>
+<p>
+Get help on the Ruby on Rails Tutorial at the
+<a href="http://www.railstutorial.org/help">Rails Tutorial
+help page</a>.
+  To get help on this sample app, see the
+<a href="http://www.railstutorial.org/book"><em>Ruby on Rails Tutorial</em>
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: use link_to helper for "a href"

File: app/views/static_pages/home.html.erb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<%= provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+This is the home page for the
+<a href="http://www.railstutorial.org/">Ruby on
```
Comment (ngocvt-0484) [inline app/views/static_pages/home.html.erb]: use link_to

File: config/routes.rb [hunk 8]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  get 'static_pages/home'
```
Comment (ngocvt-0484) [inline config/routes.rb]: use "

File: app/assets/stylesheets/custom.scss (hunk 1 above)
Comment (weilingooi) [reply app/assets/stylesheets/custom.scss]: Is this means for the function which I tab 4 space just like the second comment? Sorry, do not really understand the sentence.

**Comment by weilingooi** [comment]
ready

File: app/assets/stylesheets/application.scss [hunk 9]
```diff
@@ -13,3 +13,5 @@
  *= require_tree .
  *= require_self
  */
+ @import 'bootstrap-sprockets';
+ @import 'bootstrap';
```
Comment (saobv-2223) [inline app/assets/stylesheets/application.scss]: please add an empty line after line 17

File: app/helpers/application_helper.rb [hunk 10]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title
+    base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]: "Ruby on Rails Tutorial Sample App" you should usin I18n for this text

File: app/helpers/application_helper.rb [hunk 11]
```diff
@@ -1,2 +1,6 @@
...
+    page_title.empty? ? base_title : "#{page_title} | #{base_title}"
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]:
```suggestion
    page_title.blank? ? base_title : "#{page_title} | #{base_title}"
```

File: app/views/layouts/_header.html.erb [hunk 12]
```diff
@@ -0,0 +1,12 @@
+  <header class="navbar navbar-fixed-top navbar-inverse">
+    <div class="container">
+      <%= link_to "sample app", "#", id: "logo" %> 
+	  <nav>
+        <ul class="nav navbar-nav navbar-right"> 
+	      <li><%= link_to "Home", "#" %></li> 
+	      <li><%= link_to "Help", "#" %></li> 
+	      <li><%= link_to "Log in", "#" %></li>
+        </ul>
```
Comment (saobv-2223) [inline app/views/layouts/_header.html.erb:9]: wrong indent, only have 2 spaces for child tag html

File: app/views/layouts/application.html.erb [hunk 13]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>SampleApp</title>
+    <title><%= full_title(yield(:title)) %></title>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb]:
```suggestion
    <title><%= full_title yield(:title) %></title>
```

File: app/views/layouts/application.html.erb [hunk 14]
```diff
@@ -10,6 +10,10 @@
...
   <body>
-    <%= yield %>
+  <%= render "layouts/header" %> 
+  <div class="container">
+    <%= yield %> 
+  </div>
+	
   </body>
-</html>
+</html>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb]: Add an empty line after line 19

File: app/views/static_pages/help.html.erb [hunk 15]
```diff
@@ -0,0 +1,10 @@
+<%= provide :title, "Help" %>
+
+
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb]:
```suggestion
```

File: app/views/static_pages/help.html.erb [hunk 16]
```diff
@@ -0,0 +1,10 @@
...
+<h1>Help</h1>
+<p>
+Get help on the Ruby on Rails Tutorial at the
+<%= link_to "Rails Tutorial help page", "http://www.railstutorial.org/help"%>
+  To get help on this sample app, see the
+<%= link_to "<em>Ruby on Rails Tutorial</em>book", "http://www.railstutorial.org/book"%>
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb]: the static links, you should add to setting.yml file and call them here

File: app/views/static_pages/home.html.erb [hunk 17]
```diff
@@ -0,0 +1,8 @@
+<%= provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+This is the home page for the
+<%= link_to "Ruby on Rails Tutorial", "http://www.railstutorial.org/"%>
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]: Do same as above issue

File: app/views/static_pages/home.html.erb [hunk 18]
```diff
@@ -0,0 +1,8 @@
+<%= provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+This is the home page for the
+<%= link_to "Ruby on Rails Tutorial", "http://www.railstutorial.org/"%> 
+sample application.
+</p>
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]: Add an empty line after line 8

File: test/controllers/static_pages_controller_test.rb [hunk 19]
```diff
@@ -0,0 +1,14 @@
+require 'test_helper'
+
```
Comment (saobv-2223) [inline test/controllers/static_pages_controller_test.rb]: We don't use test folder. 
1. Add `test` to `.gitignore` file
2. Run `git rm --cached -r test`

File: app/assets/stylesheets/application.scss (hunk 9 above)
Comment (weilingooi) [reply app/assets/stylesheets/application.scss]: noted

File: app/helpers/application_helper.rb (hunk 11 above)
Comment (weilingooi) [reply app/helpers/application_helper.rb]: done

File: app/views/layouts/application.html.erb (hunk 13 above)
Comment (weilingooi) [reply app/views/layouts/application.html.erb]: done

File: app/views/static_pages/help.html.erb (hunk 15 above)
Comment (weilingooi) [reply app/views/static_pages/help.html.erb]: done

File: app/views/static_pages/home.html.erb (hunk 18 above)
Comment (weilingooi) [reply app/views/static_pages/home.html.erb]: done

File: test/controllers/static_pages_controller_test.rb (hunk 19 above)
Comment (weilingooi) [reply test/controllers/static_pages_controller_test.rb]: done

File: app/helpers/application_helper.rb (hunk 10 above)
Comment (weilingooi) [reply app/helpers/application_helper.rb]: may I know more details? or any references?

File: app/views/layouts/_header.html.erb (hunk 12 above)
Comment (weilingooi) [reply app/views/layouts/_header.html.erb:9]: header
may I know what is wrong, I haven't change any line for it but in my laptop it shown only 2 spaces for my child tag html?

File: app/views/layouts/application.html.erb (hunk 14 above)
Comment (weilingooi) [reply app/views/layouts/application.html.erb]: done

File: app/views/static_pages/help.html.erb (hunk 16 above)
Comment (weilingooi) [reply app/views/static_pages/help.html.erb]: done

File: app/views/static_pages/home.html.erb (hunk 17 above)
Comment (weilingooi) [reply app/views/static_pages/home.html.erb]: done

**Comment by weilingooi** [comment]
ready

File: app/assets/stylesheets/custom.scss (hunk 1 above)
Comment (vanvtt-0952) [reply app/assets/stylesheets/custom.scss]: yes, you should 1 tab = 2 spaces.

Current, you use 1 tab = 4 spaces. plz, change it :D

File: Gemfile [hunk 20]
```diff
@@ -37,3 +38,5 @@ group :development, :test do
   gem "rubocop", "~> 0.74.0", require: false
   gem "rubocop-rails", "~> 2.3.2", require: false
 end
+
+gem 'config'
```
Comment (vanvtt-0952) [inline Gemfile]: use `double-quotes`

File: app/controllers/static_pages_controller.rb [hunk 21]
```diff
@@ -0,0 +1,7 @@
+class StaticPagesController < ApplicationController
+  def home
+  end
```
Comment (vanvtt-0952) [inline app/controllers/static_pages_controller.rb]: this method is not body.
so, you can write it once line
example: `def home; end`

File: app/helpers/application_helper.rb [hunk 22]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title
+      base_title = t(:title)
```
Comment (vanvtt-0952) [inline app/helpers/application_helper.rb]: ` t(:title)` to `I18n.t "title"`

note:
helper file and model file: you must use `I18n.t` (full syntax of I18n)
controller file and view layer (html): you can `t ....` (sort syntax of I18n)


you can read more this link https://github.com/framgia/coding-standards/blob/master/eng/rails/standard.md

File: config/routes.rb [hunk 23]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  get "static_pages/home"
+  get "static_pages/help"
+  root "static_pages#home"
```
Comment (vanvtt-0952) [inline config/routes.rb:2]: plz arrange line code same bellow:
first line: `root "static_pages#home"`
then, After this line, have 1 empty line

then, `get "static_pages/home"`
` get "static_pages/help"`

File: Gemfile (hunk 20 above)
Comment (weilingooi) [reply Gemfile]: done

File: app/controllers/static_pages_controller.rb (hunk 21 above)
Comment (weilingooi) [reply app/controllers/static_pages_controller.rb]: done

File: app/helpers/application_helper.rb (hunk 22 above)
Comment (weilingooi) [reply app/helpers/application_helper.rb]: noted and done.

File: config/routes.rb (hunk 23 above)
Comment (weilingooi) [reply config/routes.rb:2]: done

Comment (weilingooi) [reply config/routes.rb:2]: done

**Comment by weilingooi** [comment]
ready

**Comment by vanvtt-0952** [comment]
merged
(F)

**Pull request merged**